arroyo/processing/strategies/streaming/transform.py

```diff
@@ -225,12 +231,40 @@ def parallel_transform_worker_initializer(
         custom_initialize_func()
 
 
+@dataclass
+class ParallelTransformResult(Generic[TTransformed]):
+    """
+    Results of applying the transform function to a batch
+    of messages in parallel.
+
+    `index_processed_til` holds the index of the next
+    message to be processed from the input batch. If
+    something goes wrong while processing the batch and
+    that message needs to be reprocessed, the index can
+    be passed back to the function to start processing
+    from there.
+
+    `output_batch` contains all successfully processed
+    messages.
+
+    `bad_messages` contains `InvalidMessage` objects
+    caught during processing. These messages should be
+    passed to a DLQ by raising the `InvalidMessages`
+    exception.
```

Comment:
you do not need these comments.
Call them `valid_messages` and `invalid_messages`. So that names and types say everything without the need of a redundant explanation.
At most you can add a few words to the description of the class explaining that it carries both valid messages and invalid ones.